packages/pluginutils/src/attachScopes.ts

```diff
@@ -0,0 +1,131 @@
+import { Node, walk } from 'estree-walker';
+
+import { AttachedScope, AttachScopes } from '../types';
+
+import extractAssignedNames from './extractAssignedNames';
+
+interface BlockDeclaration {
+  [index: string]: boolean;
+}
+
+const blockDeclarations: BlockDeclaration = {
```

Comment:
I think a Set would be more straightforward here.
```ts
const blockDeclarations = new Set('const let'.split(' '));
```

Reply:
Let's handle overall TS things post-migration. Looking for an analog to the current version right now.